pkg/daemon/controller/controller.go

```diff
@@ -201,17 +203,20 @@ func (sc *ServiceController) SetupWithManager(ctx context.Context, mgr ctrl.Mana
 	// This value is based on CVO's resync period
 	reconcilePeriod := 2 * time.Minute
 	eventChan := newPeriodicEventGenerator(ctx, reconcilePeriod)
+	if err := sc.ctrl.Watch(&source.Channel{Source: eventChan},
```

Comment:
Have you looked at [WatchesRawSource](https://github.com/kubernetes-sigs/controller-runtime/blob/90166996a38c148dd6941f173e55e0d17ddd4868/pkg/builder/controller.go#L179)?
It seems like that is a direct replacement for that specific call of Watches()
It can be used here but it shouldn't be used elsewhere in our code unless we have a reason to